annotation-processor/src/main/java/org/hibernate/validator/ap/util/ConstraintHelper.java

```diff
@@ -247,8 +249,8 @@ public ConstraintHelper(Types typeUtils, AnnotationApiHelper annotationApiHelper
 				BeanValidationTypes.FUTURE,
 				JAVA_TIME_TYPES_SUPPORTED_BY_FUTURE_AND_PAST_ANNOTATIONS
 		);
-		registerAllowedTypesForBuiltInConstraint( BeanValidationTypes.MAX, Number.class, String.class );
-		registerAllowedTypesForBuiltInConstraint( BeanValidationTypes.MIN, Number.class, String.class );
+		registerAllowedTypesForBuiltInConstraint( BeanValidationTypes.MAX, Number.class, String.class, Duration.class );
+		registerAllowedTypesForBuiltInConstraint( BeanValidationTypes.MIN, Number.class, String.class, Duration.class );
```

Comment:
I don't think I would add them here. We have a nice dedicated annotation, let's use that.